py/torch_tensorrt/dynamo/runtime/_TorchTensorRTModule.py

```diff
@@ -129,6 +146,40 @@ def __init__(
         if serialized_engine and not self.settings.lazy_engine_init:
             self.setup_engine()
 
+    def init_context(self):
+        self.engine.init_context()
+
+    def reset_context(self):
+        self.engine.reset_context()
+
+    def get_streamable_weights_size(self):
+        return self.engine.streamable_weights_size
+
+    def get_weight_streaming_budget(self):
+        return self.engine.weight_streaming_budget_v2
+
+    @recreate_context_decorator
+    def set_weight_streaming_budget(self, budget_bytes):
+        return self._set_weight_streaming_budget(budget_bytes)
+
+    def _set_weight_streaming_budget(self, budget_bytes):
+        # Disable weight streaming for invalid budget size
+        if budget_bytes <= 0:
+            budget_bytes = self.get_streamable_weights_size()
+
+        self.engine.weight_streaming_budget_v2 = budget_bytes
+        if self.get_weight_streaming_budget() != budget_bytes:
```

Comment:
wouldn't this always be budget_bytes since `self.engine.weight_streaming_budget_v2 = budget_bytes` is being done before this ?

Reply:
This [api](https://docs.nvidia.com/deeplearning/tensorrt/api/python_api/infer/Core/Engine.html#tensorrt.ICudaEngine) Set and get the current weight streaming budget for inference. This [test case](https://github.com/pytorch/TensorRT/pull/3111/files#diff-f9e9b91837ae150afd9f6637efb856ac105a78ec762851d0477c80a3e117dd82R141-R147) verifies it